File: misc.py
import yaml
from aocd import get_data, submit

with open('secret.yaml', 'r') as f:
    token = yaml.safe_load(f)['session']


def read_day(i: int):
    with open(f'day_{i}/input.txt', 'r', encoding='utf-8') as f:
        in_data = f.read()
    if in_data == '':
        in_data = get_data(token, day=i, year=2019)
        with open(f'day_{i}/input.txt', 'w', encoding='utf-8') as f:
            f.write(in_data)
    return in_data


def submit_day(answer, i: int, part: int):
    parts = {1: 'a', 2: 'b'}
    return submit(answer, part=parts[part], day=i, year=2019, session=token)

File: misc.jl
using DrWatson
quickactivate(@__DIR__)
using Pkg


# if Sys.iswindows()
# 	python_installation = read(`where python`, String) |> split |> x->x[1]
# 	ENV["PYTHON"] = python_installation
# 	Pkg.build("PyCall")
# elseif Sys.islinux()
# 	python_installation = read(`which python`, String) |> split |> x->x[1]
# 	ENV["PYTHON"] = python_installation
# 	Pkg.build("PyCall")
# end

using PyCall
pushfirst!(PyVector(pyimport("sys")."path"), joinpath(@__DIR__, "."))

function read_input(day::Int)
	misc = pyimport("misc")
	misc.read_day(day)
end

function submit(answer, day::Int, part::Int)
	misc = pyimport("misc")
	misc.submit_day(answer, day, part)
end

read_lines(data::AbstractString, delim='\n') = split(data, delim)
read_numbers(data::AbstractString, delim='\n') = parse.(Int, read_lines(data, delim))
read_number(data::AbstractString) = parse(Int, data)

function test_input(day::Int)
	data = open(joinpath("day_$day", "test_input.txt")) do f
		read(f, String)
	end
	data
end

function read_file(day::Int, filename)
	data = open(joinpath("day_$day", filename)) do f
		read(f, String)
	end
	data
end
